mixer/pkg/runtime/config/yaml/converter.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2018 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package yaml
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/gogo/protobuf/protoc-gen-gogo/descriptor"
+)
+
+type (
+	writer interface {
+		write(buffer *proto.Buffer) error
+	}
+	msgWriter struct {
+		field    *descriptor.FieldDescriptorProto // nil for root message
+		name     string                           // empty for root message
+		resolver Resolver
+		children []writer
+	}
+	primitiveWriter struct {
+		field    *descriptor.FieldDescriptorProto
+		data     interface{}
+		name     string
+		resolver Resolver
+	}
+	// Converter transforms yaml that represents protobuf data into []byte
+	Converter struct {
+		reader   Reader
+		resolver Resolver
+	}
+)
+
+// NewConverter creates a Converter
+func NewConverter(fds *descriptor.FileDescriptorSet) *Converter {
+	resolver := NewResolver(fds)
+	return &Converter{resolver: resolver, reader: NewReader(resolver)}
+}
+
+// ToBytes creates []byte from a yaml representation of a proto.
+func (c *Converter) ToBytes(data map[string]interface{}, msgName string, skipUnknown bool) ([]byte, error) {
```

Comment:
[ignorable] super nit: call this `Convert` (or `ConvertBytes`) or rename `Converter` to `Encoder` and have the method be `EncodeBytes`